### DISCUSSION TITLE
Revalidate anything other than `fetch()`

### DISCUSSION THREAD

**Discussion by pedro757** [vercel/next.js#49525] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #22 SMP Tue Jan 10 18:39:00 UTC 2023
    Binaries:
      Node: 16.17.0
      npm: 8.15.0
      Yarn: 1.22.19
      pnpm: 7.1.0
    Relevant packages:
      next: 13.4.2-canary.2
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/gifted-microservice-fphtss?file=%2Fpackage.json%3A26%2C25

### To Reproduce

In the codesanbox create a post and try to edit it two times you'll get staled data

### Describe the Bug

Is it possible to revalidate anything other than `fetch()` calls, like a complete path, or prisma queries.

### Expected Behavior

I expect the revalidatePath to do what the name suggests. or being able to revalidate anything other than `fetch()` calls.

For now, I have to use `router.refresh()` to get the correct data in the RSC.

**Comment by K1NXZ** [comment]
I am having the same issue. Prisma queries are not getting updated.

**Comment by timneutkens** [comment]
We're working on it! https://twitter.com/timneutkens/status/1654843969069625345

Closing as this is a question, not a bug.

**Reply by camsloanftc** [reply]
Is there anywhere to track progress on this one? Such as an RFC or similar?

**Comment by bennettdams** [comment]
I thought you can maybe use [`cache`](https://nextjs.org/docs/app/building-your-application/data-fetching/caching#react-cache) and then use [`revalidateTag`](https://nextjs.org/docs/app/building-your-application/data-fetching/revalidating#on-demand-revalidation), but turns out that `cache` does not accept a tag like `fetch` does.

**Update**:
Turns out that this is planned already! See [here](https://twitter.com/timneutkens/status/1676673959826190336) on Twitter and [here](https://github.com/vercel/next.js/blob/972d29b6d5b06cf641fb455dbb100f4d34318aa8/test/e2e/app-dir/app-static/app/variable-revalidate/revalidate-360/page.js#L20) for some code.

**Comment by aguirrealvaro** [comment]
hey @timneutkens are there news about this? would be awesome to assign a tag to a prisma query!

**Reply by ezeparziale** [reply]
With unstable_cache can cache and tag a orm query >> [unstable_cache](https://nextjs.org/docs/app/api-reference/functions/unstable_cache)
Example: https://youtu.be/VBlSe8tvg4U?si=_oC7nO3bJFV1UpBF

**Reply by aguirrealvaro** [reply]
I tried it yesterday, but the result i got was, after revalidating a tag, the whole path (page) was being revalidating
but i will take a look on the @leerob video
Thanks!
